### PR TITLE
Prepare for bigquery schema change

### LINKED ISSUE #220: Update BigQuery schema for serial_number
1. Rename diagnostics column `RPI` to `serial_number`. Column renames [aren't fully supported](https://cloud.google.com/bigquery/docs/manually-changing-schemas) so one of the workaround will have to be used.
2. Make columns nullable

Acceptance criteria
- New schema should match the result of this PR: https://github.com/NebraLtd/hm-diag/pull/219

### PR DESCRIPTION
**Pull request by ilyastrodubtsev**
**Issue**

Closes #220 

**Checklist**

- [X] Edit GCP shipper script for new schema
- [X] Edit import script in repo for support old version
- [X] Edit import script for dowloaded data from bucket to bigquery in GC Functions
- [X] Edit shema for diagnostics table in bigquery
- [X] Edit query in dashboard for support new schema
- [X]  Edit query for bigquery table "nebra-production.hotspot_diagnostics_data.most_recent_diagnostics"
- [X] Added docs about edit column name in bigquery schema

### CODE REVIEW

File: hw_diag/tests/test_gcs_shipper.py [hunk 1]
```diff
@@ -47,11 +47,11 @@ def test_upload_diagnostics_should_not_ship(self):
         retval = upload_diagnostics(diagnostics, False)
         self.assertIsNone(retval)
 
-    def test_diagnostics_to_gcs_payload(self):
+    def test_diagnostics_added_timestamp(self):
```
Comment (marvinmarnold) [inline hw_diag/tests/test_gcs_shipper.py:50]: Should prob be `test_add_timestamp_to_diagnostics`

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
Missing documentation for how to replicate the BQ table migration process.

File: Edit column names in bigquery table.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * EXCEPT(column_one, column_two),
+ column_one AS newcolumn_one, column_two AS newcolumn_two
+FROM
+ mydataset.my_new_schema_table
+```
+After creating query with your initial data. It is necessary to specify in the query settings to which table the data will be transferred. To do this, need to go to the **Query Settings**.  
+
+![img](https://user-images.githubusercontent.com/38936255/146346033-33474622-b910-4315-8eb0-cd7f2e95dd50.png)
+
+In the settings will need in the **Destination** tab:
```
Comment (vladimirpoleshko) [inline Edit column names in bigquery table.md:25]: This step is not 100% clear for me, took me a few reads to get this, maybe something like:

`Perform following steps under **Destination** tab:`

Comment (ilyastrodubtsev) [reply Edit column names in bigquery table.md]: Agree with you. It looks more understandable. Edited: https://github.com/NebraLtd/hm-diag/blob/94bf73eec3ada1aeee700751fce0a00ba53ea390/Edit%20column%20names%20in%20bigquery%20table.md?plain=1#L25

File: hw_diag/tests/test_gcs_shipper.py (hunk 1 above)
Comment (ilyastrodubtsev) [reply hw_diag/tests/test_gcs_shipper.py]: Edited: https://github.com/NebraLtd/hm-diag/blob/94bf73eec3ada1aeee700751fce0a00ba53ea390/hw_diag/tests/test_gcs_shipper.py#L50

File: Edit column names in bigquery table.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Introduction
+This is a step-by-step guide to help you change the name of a column in bigquery tables.
+
+## Step 1: Creating table with new schema in bigquery
+---
+In the **explorer** in bigquery in the dataset. Click on the ```⋮```to the right of the dataset name. And select the item **Create table**.  
+![create table](https://user-images.githubusercontent.com/38936255/146343440-b3db5c50-19b9-4314-9c7e-f82ca012d870.png)  
+And create a new table using the schema with the changed name of the field or fields.
```
Comment (kashifpk) [inline Edit column names in bigquery table.md:8]: And=and

File: Edit column names in bigquery table.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# Introduction
+This is a step-by-step guide to help you change the name of a column in bigquery tables.
+
+## Step 1: Creating table with new schema in bigquery
+---
+In the **explorer** in bigquery in the dataset. Click on the ```⋮```to the right of the dataset name. And select the item **Create table**.  
+![create table](https://user-images.githubusercontent.com/38936255/146343440-b3db5c50-19b9-4314-9c7e-f82ca012d870.png)  
+And create a new table using the schema with the changed name of the field or fields.  
+At the end of the first step, you should have two tables. One original, the second with a modified scheme.
```
Comment (kashifpk) [inline Edit column names in bigquery table.md:9]: schema=schema (at the end)

File: Edit column names in bigquery table.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+4. Select in **Destination table write preference** action **Append to table**.  
+5. Click **Save**.
+
+![img](https://user-images.githubusercontent.com/38936255/146348178-978cb3ae-64dd-43a1-b236-c6f9e7007de2.png)  
+
+After that, run the query and wait for it to complete successfully.
+
+## Step 3: Replacing an old table with a new
+---
+*Recommend: Before proceeding, make a backup copy of the current table to protect yourself.*
```
Comment (kashifpk) [inline Edit column names in bigquery table.md:38]: Recommend=Recommended

File: Edit column names in bigquery table.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+This is a step-by-step guide to help you change the name of a column in bigquery tables.
+
+## Step 1: Creating table with new schema in bigquery
+---
+In the **explorer** in bigquery in the dataset. Click on the ```⋮```to the right of the dataset name. And select the item **Create table**.  
+![create table](https://user-images.githubusercontent.com/38936255/146343440-b3db5c50-19b9-4314-9c7e-f82ca012d870.png)  
+And create a new table using the schema with the changed name of the field or fields.  
+At the end of the first step, you should have two tables. One original, the second with a modified scheme.  
+![img](https://user-images.githubusercontent.com/38936255/146344509-2aecf8bb-119c-455b-ba44-a95924bcc532.png)  
+## Step 2:  Import data from a table using the old schema into a new
```
Comment (kashifpk) [inline Edit column names in bigquery table.md:11]: May be clarify here that we are importing data from table with old schema to a new (3rd) table or the new table we created with the new schema.

File: Edit column names in bigquery table.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
+# Introduction
```
Comment (kashifpk) [inline Edit column names in bigquery table.md:1]: May be add some context here about why we need this procedure. Reading this document it looks like we cannot simply rename fields or tables in big query? is this procedure for solving this restriction?

File: Edit column names in bigquery table.md (hunk 3 above)
Comment (ilyastrodubtsev) [reply Edit column names in bigquery table.md]: Create this more clean: https://github.com/NebraLtd/hm-diag/blob/0fe933265d2af405f6ae204e470fff3469f716f1/Edit%20column%20names%20in%20bigquery%20table.md?plain=1#L11-L14

File: Edit column names in bigquery table.md (hunk 4 above)
Comment (ilyastrodubtsev) [reply Edit column names in bigquery table.md]: Fixed typo: https://github.com/NebraLtd/hm-diag/blob/0fe933265d2af405f6ae204e470fff3469f716f1/Edit%20column%20names%20in%20bigquery%20table.md?plain=1#L14

File: Edit column names in bigquery table.md (hunk 5 above)
Comment (ilyastrodubtsev) [reply Edit column names in bigquery table.md]: Fixed: https://github.com/NebraLtd/hm-diag/blob/0fe933265d2af405f6ae204e470fff3469f716f1/Edit%20column%20names%20in%20bigquery%20table.md?plain=1#L43

File: Edit column names in bigquery table.md (hunk 6 above)
Comment (ilyastrodubtsev) [reply Edit column names in bigquery table.md]: Make this more understandable: https://github.com/NebraLtd/hm-diag/blob/0fe933265d2af405f6ae204e470fff3469f716f1/Edit%20column%20names%20in%20bigquery%20table.md?plain=1#L16

File: Edit column names in bigquery table.md (hunk 7 above)
Comment (ilyastrodubtsev) [reply Edit column names in bigquery table.md:1]: Great idea @kashifpk
I Did this.
https://github.com/NebraLtd/hm-diag/blob/0fe933265d2af405f6ae204e470fff3469f716f1/Edit%20column%20names%20in%20bigquery%20table.md?plain=1#L1-L7

**Review by marvinmarnold — APPROVED** [review]

**Pull request merged**